File: app/rater/forecastRatings.m
function [tTree mTree] = forecastRatings(isOptimized, isUpdated)
  setup;
  [tTree mTree T winTiesRatio] = readData(dataPath);
  mi = MatchIterator(mTree);
  [tTree mTree mi] = optimizeRatings(tTree, mTree, mi, ...
      winTiesRatio, isOptimized);
  
  if (isUpdated)
    T = writeData(mTree, mi, T);
  end
end 

File: odmsRater/main/forecastRatings.R
forecastRatings <- function(currentDate, rData=NULL) {
  library(MASS)
  library(hash)
  library(parallel)
  regexRFiles <- ".*\\.R"
  srcFiles <- list.files("../", regexRFiles,
      full.names=TRUE, recursive=TRUE)
  utilitiesFiles <- list.files("../../utilities", regexRFiles,
      full.names=TRUE, recursive=TRUE)
  sapply(c(srcFiles, utilitiesFiles), source)
  dateFormat <- "%m/%d/%y"
  dataPath <- "../../data/"
  currentDate <- as.Date(currentDate, dateFormat)
  readsData <- readData(currentDate, dateFormat, dataPath)
  tTree <- readsData[["tTree"]]
  fTree <- readsData[["fTree"]]
  gTree <- readsData[["gTree"]]
  T <- readsData[["T"]]
  gi <- newGameIterator(gTree)
  rData <- optimizeRatings(tTree, fTree, gTree, gi, rData)
  writeData(rData, T, dataPath)
  rData
}
